libfeed/feed-text-sanitization.c

```diff
@@ -0,0 +1,118 @@
+/* Copyright 2018 Endless Mobile, Inc.
+ *
+ * eos-discovery-feed is free software: you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 2.1 of the
+ * License, or (at your option) any later version.
+ *
+ * eos-discovery-feed is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with eos-discovery-feed.  If not, see
+ * <http://www.gnu.org/licenses/>.
+ */
+
+#include <string.h>
+
+#include <glib.h>
+
+#include "feed-text-sanitization.h"
+
+static gchar *
+regex_replace (const gchar *exp, const gchar *str, const gchar *replacement)
+{
+  g_autoptr(GError) error = NULL;
+  g_autoptr(GRegex) regex = g_regex_new (exp, 0, 0, &error);
+  g_autofree gchar *replaced = NULL;
+
+  if (regex == NULL)
+    {
+      g_error ("Fatal error in compiling regular expression: %s", error->message);
+      return NULL;
+    }
+
+  replaced = g_regex_replace (regex, str, -1, 0, replacement, 0, &error);
+
+  if (replaced == NULL)
+    {
+      g_error ("Fatal error in performing string replacement: %s", error->message);
+      return NULL;
+    }
+
+  return g_steal_pointer (&replaced);
+}
+
+static gchar *
+remove_square_brackets (const gchar *str)
+{
+  return regex_replace ("\\[\\d+\\]", str, "");
+}
+
+static gchar *
+remove_parens (const gchar *str)
+{
+  return regex_replace ("\\(.*?\\)", str, "");
+}
+
+static gchar *
+trim_to_first_two_sentences (const gchar *str)
+{
+  g_autoptr(GPtrArray) sentences = g_ptr_array_new_with_free_func (g_free);
+  g_auto(GStrv) all_sentences = g_strsplit (str, ".", -1);
+  GStrv iter = all_sentences;
+  guint sentence_count = 0;
+
+  for (; *iter != NULL && sentence_count < 2; ++iter)
+    {
+      if (strlen (*iter) > 0)
+        g_ptr_array_add (sentences, g_strdup (*iter));
```

Comment:
Since `all_sentences` is valid for the whole lifecycle of this function and `g_strjoinv()` at the end will make another copy, I'm pretty sure you don't need to copy the strings here.

Reply:
Handy, I'll remove the copy :)